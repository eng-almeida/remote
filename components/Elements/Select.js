import PropTypes from 'prop-types';
import styled from 'styled-components';

import Field from './Field';
import { FieldText } from './Typography';

const SelectContainer = styled.div`
  position: relative;
`;

const Select = styled.select`
  ${FieldText}
  border: none;
  border-bottom: 1px solid ${({ theme }) => theme.colors.grays[400]};
  color: ${({ theme }) => theme.colors.grays[300]};
  outline: none;
  padding-bottom: ${({ theme }) => theme.space[2]};
  width: 100%;

  -moz-appearance: none;
  -webkit-appearance: none;
  &::-ms-expand {
    display: none;
  }
`;

const Caret = styled.span`
  position: absolute;
  right: ${({ theme }) => theme.space[4]};
  top: ${({ theme }) => theme.space[2]};
  display: inline-flex;
  width: 10px;
  height: 10px;
  border-left: 2px solid ${({ theme }) => theme.colors.grays[400]};
  border-bottom: 2px solid ${({ theme }) => theme.colors.grays[400]};
  transform: rotate(-45deg);
`;

const SelectComponent = ({ label, helperText, children, ...inputProps }) => (
  <Field label={label} helperText={helperText} id={inputProps.id}>
    <SelectContainer>
      <Select {...inputProps}>{children}</Select>
      <Caret />
    </SelectContainer>
  </Field>
);

SelectComponent.propTypes = {
  label: PropTypes.string,
  helperText: PropTypes.string,
  children: PropTypes.arrayOf(PropTypes.element),
};

export default SelectComponent;
