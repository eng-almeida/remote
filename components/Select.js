import styled from 'styled-components';

import Field from './Field';

const Select = styled.select`
	border: none;
	border-bottom: 1px solid ${({ theme }) => theme.colors.grays[400]};
	color: ${({ theme }) => theme.colors.grays[300]};
	font-size: 20px;
	line-height: 24px;
	outline: none;
	padding-bottom: ${({ theme }) => theme.space[2]};
	width: 100%;

	-moz-appearance: none; 
	-webkit-appearance: none; 
	&::-ms-expand { 
		display: none; 
	}
`;

const SelectComponent = ({ label, helperText, children, ...inputProps }) => {
	return (
		<Field label={label} helperText={helperText} id={inputProps.id}>
			<Select {...inputProps}>
				{ children }
			</Select>
		</Field>
	);
};

export default SelectComponent;