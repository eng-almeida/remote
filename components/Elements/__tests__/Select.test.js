import Select from '@/components/Elements/Select';

describe('Select component', () => {
	it('should match snapshot', () => {
		const { container } = renderWithProviders(
			<Select 
				id="test"
				name="test"
				label="Test"
				helperText="Test helper"
      >
      	<option value="Test 1">Test 1</option>
      	<option value="Test 2">Test 2</option>
      	<option value="Test 3">Test 3</option>
			</Select>
		);
		expect(container).toMatchSnapshot();
	});
})