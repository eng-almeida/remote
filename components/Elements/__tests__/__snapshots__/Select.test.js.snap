// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select component should match snapshot 1`] = `
.c0 {
  font-size: 13px;
  line-height: 16px;
  display: block;
  color: #778CA3;
  margin-bottom: 4px;
}

.c4 {
  font-size: 13px;
  line-height: 16px;
  color: #00234B;
  opacity: 0.5;
  margin-top: 8px;
}

.c1 {
  position: relative;
}

.c2 {
  font-size: 20px;
  line-height: 24px;
  border: none;
  border-bottom: 1px solid #CED5DB;
  color: #00234B;
  outline: none;
  padding-bottom: 8px;
  width: 100%;
  -moz-appearance: none;
  -webkit-appearance: none;
}

.c2::-ms-expand {
  display: none;
}

.c3 {
  position: absolute;
  right: 16px;
  top: 8px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  width: 10px;
  height: 10px;
  border-left: 2px solid #CED5DB;
  border-bottom: 2px solid #CED5DB;
  -webkit-transform: rotate(-45deg);
  -ms-transform: rotate(-45deg);
  transform: rotate(-45deg);
}

<div>
  <div>
    <label
      class="c0"
      for="test"
    >
      Test
    </label>
    <div
      class="c1"
    >
      <select
        class="c2"
        id="test"
        name="test"
      >
        <option
          value="Test 1"
        >
          Test 1
        </option>
        <option
          value="Test 2"
        >
          Test 2
        </option>
        <option
          value="Test 3"
        >
          Test 3
        </option>
      </select>
      <span
        class="c3"
      />
    </div>
    <div
      class="c4"
    >
      Test helper
    </div>
  </div>
</div>
`;
